ecommerce/extensions/iap/api/v1/views.py

```diff
@@ -0,0 +1,95 @@
+# pylint: disable=no-else-return
+
+
+import logging
+import time
+from django.http import JsonResponse
+from django.utils.html import escape
+from django.utils.translation import ugettext as _
+from edx_rest_framework_extensions.permissions import LoginRedirectIfUnauthenticated
+from oscar.apps.basket.views import *  # pylint: disable=wildcard-import, unused-wildcard-import
+from rest_framework.views import APIView
+from ecommerce.extensions.analytics.utils import track_segment_event
+from ecommerce.extensions.basket.constants import EMAIL_OPT_IN_ATTRIBUTE
+from ecommerce.extensions.basket.exceptions import BadRequestException, RedirectException
+from ecommerce.extensions.basket.utils import prepare_basket
+from ecommerce.extensions.basket.views import BasketLogicMixin
+from ecommerce.extensions.order.exceptions import AlreadyPlacedOrderException
+from ecommerce.extensions.partner.shortcuts import get_partner_for_site
+
+BasketAttribute = get_model('basket', 'BasketAttribute')
+BasketAttributeType = get_model('basket', 'BasketAttributeType')
+logger = logging.getLogger(__name__)
+Product = get_model('catalogue', 'Product')
+Voucher = get_model('voucher', 'Voucher')
+
+
+class MobileBasketAddItemsView(BasketLogicMixin, APIView):
+    """
+    View that adds multiple products to a mobile user's basket.
+    """
+    permission_classes = (LoginRedirectIfUnauthenticated,)
+
+    def get(self, request):
+        # Send time when this view is called - https://openedx.atlassian.net/browse/REV-984
+        properties = {'emitted_at': time.time()}
+        track_segment_event(request.site, request.user, 'Basket Add Items View Called', properties)
+
+        try:
+            skus = self._get_skus(request)
+            products = self._get_products(request, skus)
+
+            logger.info('Starting payment flow for user [%s] for products [%s].', request.user.username, skus)
+
+            available_products = self._get_available_products(request, products)
+
+            try:
+                basket = prepare_basket(request, available_products)
+            except AlreadyPlacedOrderException:
+                return JsonResponse({'error': _('You have already purchased these products')}, status=406)
```

Comment:
406 doesn't seem quite right to me, since this isn't related to the request's `Accept` header value.  Is this copied from elsewhere?  If so I guess let's leave it, but otherwise maybe it should be a 422?

Reply:
Yeah since it is a processing error I think a `422` would be more appropriate. Thanks for pointing out